new/prusti-specs/src/specifications/untyped.rs

```diff
@@ -169,6 +170,9 @@ impl AssignExpressionId<Assertion> for common::Assertion<(), syn::Expr, Arg> {
 /// A trait for encoding the statements for type-checking the spec.
 pub trait EncodeTypeCheck {
     fn encode_type_check(&self, tokens: &mut TokenStream);
+    fn encode_type_check_any_type(&self, tokens: &mut TokenStream) {
```

Comment:
Could you add doc comments to both methods and also explain why it is fine to have an explicit panic here?

Reply:
I introduced this because of two types of expressions: one of them has to be bools and the other one not (triggers) so i need two ways to encode; the second one is relevant only for expressions (i am not sure if it will always be the case), therefore i picked a default implementation

Reply:
Actually, I thought a bit and I think it would be cleaner to have a separate type for trigger instead of `Expression` so that you can implement a custom `encode_type_check` and avoid `encode_type_check_any_type`.